### PR TITLE
Add receipt types constant

### PR DESCRIPTION
**Pull request by clokep**
Trying to get a bit familiar with the receipts code, figured I would:

* Add a constant for `m.read`.
* Finish up some missing type hints.

### CODE REVIEW

**Comment by clokep** [comment]
There's a few spots where we add/remove fields from JSON objects with the `m.read` flag -- it is unclear to me if those should use the same constant though. Any thoughts?

**Comment by richvdh** [comment]
> There's a few spots where we add/remove fields from JSON objects with the m.read flag -- it is unclear to me if those should use the same constant though. Any thoughts?

I'd have thought we should - they're referring to the same concept? Do you have examples?

File: synapse/storage/databases/main/receipts.py [hunk 1]
```diff
@@ -78,7 +89,7 @@ def __init__(self, database: DatabasePool, db_conn, hs: "HomeServer"):
             "ReceiptsRoomChangeCache", self.get_max_receipt_stream_id()
         )
 
-    def get_max_receipt_stream_id(self):
+    def get_max_receipt_stream_id(self) -> int:
         """Get the current max stream ID for receipts stream
 
         Returns:
```
Comment (richvdh) [inline synapse/storage/databases/main/receipts.py:95]: might be nice to remove this

**Review by richvdh — APPROVED** [review]
lgtm!

**Comment by clokep** [comment]
> > There's a few spots where we add/remove fields from JSON objects with the m.read flag -- it is unclear to me if those should use the same constant though. Any thoughts?
> 
> I'd have thought we should - they're referring to the same concept? Do you have examples?

The REST layer pulls some bits out, for example:

https://github.com/matrix-org/synapse/blob/086adf55f9b3624853977eb470fb6b61faec32ce/synapse/rest/client/read_marker.py#L51

I think we also generate JSON payloads with the field for sync (and maybe push).

It seems there's also the `ReadReceiptEventFields` which I could possibly add something too...

**Comment by richvdh** [comment]
> The REST layer pulls some bits out, for example:
> 
> https://github.com/matrix-org/synapse/blob/086adf55f9b3624853977eb470fb6b61faec32ce/synapse/rest/client/read_marker.py#L51

That certainly looks like it should be using the same constant to me.

File: synapse/storage/databases/main/receipts.py (hunk 1 above)
Comment (clokep) [reply synapse/storage/databases/main/receipts.py]: Good catch! I found one other too!

**Comment by clokep** [comment]
> > The REST layer pulls some bits out, for example:
> > https://github.com/matrix-org/synapse/blob/086adf55f9b3624853977eb470fb6b61faec32ce/synapse/rest/client/read_marker.py#L51
> 
> That certainly looks like it should be using the same constant to me.

I've updated a few more places. 👍

**Pull request merged**